extension-testcontainer/src/main/java/io.camunda.zeebe.process.test.extension.testcontainer/ZeebeProcessTestExtension.java

```diff
@@ -193,4 +191,21 @@ private Optional<ObjectMapper> getCustomMapper(final ExtensionContext context) {
       throw new RuntimeException(e);
     }
   }
+
+  /**
+   * Create a {@link ZeebeClient}. If a custom {@link ObjectMapper} is provided it is initialized into
+   * the {@link ObjectMapperConfig} and it is configured into the client
+   * @param objectMapper an {@link Optional} of {@link ObjectMapper}
+   * @param engine the used engine
+   * @return a zeebe client
+   */
+  private ZeebeClient createClient(
+      final Optional<ObjectMapper> objectMapper, final ContainerizedEngine engine) {
```

Comment:
🔧 Optional shouldn't be passed as a parameter. This also made me realise we can make this code a lot more clean.

In the `getCustomMapper` method we can always return an `ObjectMapper`. If the user configured one we return this one, else we return `new ObjectMapper()`. This gets rid of all these checks and behaves the same as now, as an empty ObjectMapper is the default our client uses.